kansha/board/comp.py

```diff
@@ -263,25 +258,29 @@ def create_column(self, index, title, nb_cards=None, archive=False):
             return False
         col = DataColumn.create_column(self.data, index, title, nb_cards, archive=archive)
         if not archive or (archive and self.archive):
-            self.columns.insert(index, component.Component(self._services(column.Column, col.id, self, self.search_engine), 'new'))
+            self.columns.insert(index, component.Component(self._services(column.Column, col.id, self), 'new'))
         self.increase_version()
         return col.id
 
-    def delete_column(self, id_):
+    def delete_column(self, col):
         """Delete a board's column
 
         In:
-            - ``id_`` -- the id of the column to delete
+            - ``col`` -- the column business object
         """
-
+        # TODO: create_column should return a column business object for consistency
         security.check_permissions('edit', self)
-        for comp in self.columns:
-            if comp().data.id == id_:
-                self.columns.remove(comp)
-                comp().delete()
-                self.increase_version()
-                return popin.Empty()
-        raise exceptions.KanshaException('No column with id [%s] found' % id_)
+        for col_comp in self.columns:
+            if col_comp() is col:
+                break
+        else:
+            raise exceptions.KanshaException('No column with id [%s] found' % col.db_id)
+
+        self.columns.remove(col_comp)
+        self.archive_card(col.get_cards(), col)
+        self.data.delete_column(col.data)
+        self.increase_version()
+        return popin.Empty()
```

Comment:
Important: the business actions (here the removal of a card) must be separated from the UI actions (here, to close a popin)